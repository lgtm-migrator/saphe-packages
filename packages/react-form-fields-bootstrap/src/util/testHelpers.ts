import useForm, { Field } from '@saphe/react-form';
import { renderHook } from '@testing-library/react-hooks';
import { ReactElement } from 'react';
import { BootstrapFieldPack } from '..';

interface FormState {
  Form: () => ReactElement;
}

export const renderForm = (hook: () => FormState): (() => ReactElement) =>
  renderHook(hook).result.current.Form;

export const minimalTextFieldForm = (props?: {
  description?: string;
}): FormState =>
  useForm({
    name: 'minimalTextFieldForm',
    fieldPack: BootstrapFieldPack,
    fields: {
      textField: {
        type: Field.TEXT,
        label: 'Text Field',
        description: props?.description,
        // validation: {
        //   required:
        // }
      },
    },
  });

export const minimalTextAreaFieldForm = (props?: {
  description?: string;
}): FormState =>
  useForm({
    name: 'minimalTextAreaFieldForm',
    fieldPack: BootstrapFieldPack,
    fields: {
      textAreaField: {
        type: Field.TEXT_AREA,
        label: 'Textarea Field',
        description: props?.description,
      },
    },
  });

export const minimalSelectFieldForm = (props?: {
  description?: string;
}): FormState =>
  useForm({
    name: 'minimalSelectFieldForm',
    fieldPack: BootstrapFieldPack,
    fields: {
      selectField: {
        type: Field.SELECT,
        label: 'Select Field',
        description: props?.description,
        options: [
          {
            label: 'Option 1',
            value: 'option-1',
          },
          {
            label: 'Option 2',
            value: 'option-2',
          },
        ],
      },
    },
  });

export const minimalCheckBoxFieldForm = (props?: {
  description?: string;
}): FormState =>
  useForm({
    name: 'minimalCheckBoxFieldForm',
    fieldPack: BootstrapFieldPack,
    fields: {
      checkBoxField: {
        type: Field.CHECK,
        label: 'Checkbox Field',
        description: props?.description,
      },
    },
  });

export const minimalNumberFieldForm = (props?: {
  description?: string;
}): FormState =>
  useForm({
    name: 'minimalNumberFieldForm',
    fieldPack: BootstrapFieldPack,
    fields: {
      numberField: {
        type: Field.NUMBER,
        label: 'Number Field',
        description: props?.description,
      },
    },
  });
