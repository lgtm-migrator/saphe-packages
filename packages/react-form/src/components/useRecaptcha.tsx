import React, { ReactElement, useCallback, useState } from 'react';
import RecaptchaComponent from 'react-google-recaptcha';

export interface RecaptchaConfig {
  siteKey: string;
  locale: string;
  errorMessage: string;
}

export interface State {
  Recaptcha: () => ReactElement;
  recaptchaToken: string | null;
}

function useRecaptcha(config?: RecaptchaConfig): State {
  const [token, setToken] = useState<string | null>(null);

  const handleChange = (newToken: string | null) => setToken(newToken);
  const handleExpired = () => setToken(null);
  const handleErrored = () =>
    console.error('Network Disconnected, cannot verify reCAPTCHA');

  const Recaptcha = useCallback(
    () => (
      <div className="mb-3">
        {!!config && (
          <RecaptchaComponent
            sitekey={config.siteKey}
            hl={config.locale}
            onChange={handleChange}
            onExpired={handleExpired}
            onErrored={handleErrored}
          />
        )}
      </div>
    ),
    [config],
  );

  return { Recaptcha, recaptchaToken: token };
}

export default useRecaptcha;
