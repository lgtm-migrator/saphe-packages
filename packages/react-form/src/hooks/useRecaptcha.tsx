import React, { ReactElement, useCallback, useState } from 'react';
import RecaptchaComponent from 'react-google-recaptcha';

export interface Config {
  siteKey: string;
  locale: string;
  onError: () => void;
}

export interface State {
  Recaptcha: () => ReactElement;
  recaptchaToken: string | undefined;
}

export default function useRecaptcha(config?: Config): State {
  const [token, setToken] = useState<string | undefined>(undefined);

  const handleChange = (newToken: string | null) =>
    setToken(newToken ?? undefined);
  const handleExpired = () => setToken(undefined);
  const handleErrored = () =>
    console.error('Network Disconnected, cannot verify reCAPTCHA');

  const Recaptcha = useCallback(
    () =>
      config ? (
        <div>
          <RecaptchaComponent
            sitekey={config.siteKey}
            hl={config.locale}
            onChange={handleChange}
            onExpired={handleExpired}
            onErrored={handleErrored}
          />
        </div>
      ) : (
        <div />
      ),
    [config],
  );

  return { Recaptcha, recaptchaToken: token };
}
