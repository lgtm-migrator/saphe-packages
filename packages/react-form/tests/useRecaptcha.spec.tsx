import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import useRecaptcha from '../src/hooks/useRecaptcha';
import { matchSnapshot } from './testHelpers';

describe('useRecaptcha', () => {
  it('renders', async () => {
    const {
      result: {
        current: { Recaptcha, recaptchaToken },
      },
    } = renderHook(() =>
      useRecaptcha({
        onError: () => alert('Recaptcha error message'),
        locale: 'en',
        siteKey: 'siteKey',
      }),
    );

    matchSnapshot(<Recaptcha />);
    expect(recaptchaToken).toBeUndefined();
  });

  it("doesn't render without config", () => {
    const {
      result: {
        current: { Recaptcha, recaptchaToken },
      },
    } = renderHook(() => useRecaptcha(undefined));
    matchSnapshot(<Recaptcha />);
    expect(recaptchaToken).toBeUndefined();
  });
});
