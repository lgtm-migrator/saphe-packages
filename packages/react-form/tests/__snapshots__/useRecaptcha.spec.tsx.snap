// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useRecaptcha doesn't render without config 1`] = `null`;

exports[`useRecaptcha renders 1`] = `<div />`;
